api/app/Services/ModelFactoryInstance.php

```diff
@@ -248,6 +248,12 @@ public function transformed()
             }
 
             $this->setEntityType();
+        } elseif ($this->factoryInstance instanceof BaseModel) {
+            $entity = $this->factoryInstance;
```

Comment:
Also didn't like that violition of single responsibility principle.
Not sure if it will not break any template, as this was added for Collection for some reason.

Can give a try removing it =)